runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -55,6 +56,22 @@ public BackpointerStreamView(final CorfuRuntime runtime,
         this.options = StreamOptions.DEFAULT;
     }
 
+    @Override
+    public void gc(long trimMark) {
+        NavigableSet<Long> q1 = getCurrentContext().readCpQueue;
+        NavigableSet<Long> q2 = getCurrentContext().readQueue;
+        NavigableSet<Long> q3 = getCurrentContext().resolvedQueue;
+        // TODO(Maithem): set slicing might be more efficiency than this
```

Comment:
Yes, you can explore using something like
`resolvedQueue.headSet(trimMark).clear();`

Reply:
Done.